app/controllers/concerns/sufia/files_controller_behavior.rb

```diff
@@ -64,6 +64,7 @@ def load_resource_from_solr
 
     # routed to /files/new
     def new
+      set_variables_for_new_form
```

Comment:
I'm adding two more tests that cover the scenario where a user tries to add a file to a collection they cannot edit.